packages/component-library-angular/src/button-link/ButtonLink.spec.ts

```diff
@@ -0,0 +1,190 @@
+import { render, screen } from '@testing-library/angular';
+import '@testing-library/jest-dom';
+import { UtrechtButtonLinkAttr } from './component';
+
+describe('Button Link', () => {
+  it('renders a button link role element', async () => {
+    await render('<a utrecht-button-link href="https://example.com">Link</a>', {
+      declarations: [UtrechtButtonLinkAttr],
+    });
+
+    const buttonLink = screen.getByRole('link', { name: 'Link' });
+
+    expect(buttonLink).toBeInTheDocument();
+    expect(buttonLink).toBeVisible();
+  });
+
+  it('renders an HTML button link element', async () => {
+    const { container } = await render('<a utrecht-button>Read more</a>', {
+      declarations: [UtrechtButtonLinkAttr],
+    });
+
+    const buttonLink = container.querySelector('a:only-child');
+
+    expect(buttonLink).toBeInTheDocument();
+  });
+
+  it('renders button link that contain HTML rich text content', async () => {
+    const { container } = await render(
+      `<a href="https://example.com" utrecht-button-link>
+          Visit <strong>now</strong>
+        </a>`,
+      {
+        declarations: [UtrechtButtonLinkAttr],
+      },
+    );
+    const buttonLink = container.querySelector(':only-child');
+
+    const richText = buttonLink?.querySelector('strong');
+
+    expect(buttonLink).toContainElement(richText as HTMLElement);
+  });
+
+  it('renders a design system BEM class name', async () => {
+    const { container } = await render('<a href="https://example.com" utrecht-button-link></a>', {
+      declarations: [UtrechtButtonLinkAttr],
+    });
+
+    const buttonLink = container.querySelector(':only-child');
+
+    expect(buttonLink).toHaveClass('utrecht-button-link');
+    expect(buttonLink).toHaveClass('utrecht-button-link--html-a');
+  });
+
+  it('can be hidden', async () => {
+    const { container } = await render('<a href="https://example.com" [hidden]="hidden" utrecht-button-link></a>', {
+      declarations: [UtrechtButtonLinkAttr],
+      componentProperties: {
+        hidden: true,
+      },
+    });
+
+    const buttonLink = container.querySelector(':only-child');
+
+    expect(buttonLink).not.toBeVisible();
+  });
+
+  it('can have a addition class name', async () => {
+    const { container } = await render('<a href="https://example.com" [class]="class" utrecht-button-link></a>', {
```

Comment:
Is there a reason to use `[class]="class"` instead of simply `class="custom-class"`? The code would probably be a bit more straightforward to understand. Same goes for `hidden`, I suppose.

Reply:
Actually, it works fine without `[class]="class"` also easier, the reason why I used it, is I thought it was like when we export a custom tag like `<utrecht-button-link />`